.github/workflows/pypi_upload.yml

```diff
@@ -0,0 +1,78 @@
+name: upload-pypi
```

Comment:
This actually might be good to breakout to `pytorch/test-infra` eventually, seems generally useful

Reply:
Yeah, let me do that.  Moving this to test-infra would actually solve the other comments you have about the complexity of the input matrix.  I only need the matrix to feed it into `./test-infra/.github/actions/setup-binary-upload`, the one that set `env.ARTIFACT_NAME`.  For the context, this setup allows the upload to go on even when some build config fails

Reply:
Here it goes https://github.com/pytorch/test-infra/pull/5217